sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -287,16 +294,44 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    * Checks if the database with the specified name exists.
    */
   override def databaseExists(dbName: String): Boolean = {
-    sessionCatalog.databaseExists(dbName)
+    // To maintain backwards compatibility, we first treat the input is a simple dbName and check
+    // if sessionCatalog contains it. If no, we try to parse it as 3 part name. If the parased
+    // identifier contains both catalog name and database name, we then search the database in the
+    // catalog.
+    if (!sessionCatalog.databaseExists(dbName)) {
+      val ident = sparkSession.sessionState.sqlParser.parseMultipartIdentifier(dbName)
+      if (ident.length == 2) {
```

Comment:
hmm shall we consider multi-level namespace like `cat.ns1.ns2`?

Reply:
hmm I don't know how to deal with nested namespace without the help of SQL analyzer. I am using `sparkSession.sessionState.catalogManager.catalog(catalog_name)` to check if a database exists.


Maybe we should add a V2 command to handle nested namespace? If I handle by calling `catalogManager`, it seems not to be extensible?

Reply:
We can follow what we did in `makeTable`: Create a `UnresolvedNamespace` logical plan and resolve it.